platforms/core-runtime/build-profile/src/main/java/org/gradle/profile/BuildProfileServices.java

```diff
@@ -40,7 +40,7 @@ public void registerBuildServices(ServiceRegistration registration) {
         registration.addProvider(new ServiceRegistrationProvider() {
             public void configure(ServiceRegistration serviceRegistration, StartParameter startParameter) {
                 if (startParameter.isProfile()) {
-                    serviceRegistration.add(ProfileEventAdapter.class);
+                    serviceRegistration.add(ProfileService.class, ProfileEventAdapter.class);
```

Comment:
Don't accidentally register the listener interfaces that the `ProfileEventAdapter` implements as services.